src/main/java/rx/schedulers/ScheduleWhen.java

```diff
@@ -0,0 +1,248 @@
+ * 		// delay the starting of the next worker by 1 second.
+ * 		return worker.delay(1, TimeUnit.SECONDS);
+ * 	}));
+ * }, Schedulers.computation());
+ * </pre>
+ * <p>
+ * Limit the amount concurrency two at a time without creating a new fix size
+ * thread pool:
+ * 
+ * <pre>
+ * Scheduler limitSched = new ScheduleWhen(workers -> {
+ * 	// use merge max concurrent to two at a time
+ * 	return Completable.merge(workers, 2);
+ * }, Schedulers.computation());
+ * </pre>
+ */
+@Experimental
+public class ScheduleWhen extends Scheduler {
+	private final Scheduler actualScheduler;
+	private final PublishSubject<Completable> workerQueue;
+
+	public ScheduleWhen(Func1<Observable<Completable>, Completable> combine, Scheduler actualScheduler) {
+		this.actualScheduler = actualScheduler;
+		// workers are converted into completables and put in this queue.
+		this.workerQueue = PublishSubject.create();
+		// send it to a custom combinator to pick the order and rate at which
+		// workers are processed.
+		combine.call(workerQueue.onBackpressureBuffer()).subscribe();
+	}
+
+	@Override
+	public Worker createWorker() {
+		// a queue for the actions submitted while worker is waiting to get to
+		// the subscribe to off the workerQueue.
+		final BufferUntilSubscriber<ScheduledAction> actionQueue = BufferUntilSubscriber.create();
+		final Worker actualWorker = actualScheduler.createWorker();
+
+		// convert the work of scheduling all the actions into a completable
+		Completable completable = actionQueue.doOnNext(new Action1<ScheduledAction>() {
+			@Override
+			public void call(ScheduledAction action) {
+				action.call(actualWorker);
+			}
+		}).toCompletable();
+
+		// a worker that queues the action to the actionQueue subject.
+		Worker worker = new Worker() {
+			private final AtomicBoolean unsubscribed = new AtomicBoolean();
+
+			@Override
+			public void unsubscribe() {
+				// complete the actionQueue when worker is unsubscribed to make
+				// room for the next worker in the workerQueue.
+				if (unsubscribed.compareAndSet(false, true)) {
+					actualWorker.unsubscribe();
+					actionQueue.onCompleted();
+				}
+			}
+
+			@Override
+			public boolean isUnsubscribed() {
+				return unsubscribed.get();
+			}
+
+			@Override
+			public Subscription schedule(final Action0 action, final long delayTime, final TimeUnit unit) {
+				// send a scheduled action to the actionQueue
+				DelayedAction delayedAction = new DelayedAction(action, delayTime, unit);
+				actionQueue.onNext(delayedAction);
+				return delayedAction;
+			}
+
+			@Override
+			public Subscription schedule(final Action0 action) {
+				// send a scheduled action to the actionQueue
+				ImmediateAction immediateAction = new ImmediateAction(action);
+				actionQueue.onNext(immediateAction);
+				return immediateAction;
+			}
+		};
+
+		// enqueue the completable that process actions put in reply subject
+		workerQueue.onNext(completable);
```

Comment:
Workers can be created from multiple threads and this will likely mess up the internal state of the `PublishSubject`+`onBackpressureBuffer`.